diff --git a/thirdparty/embree/common/math/emath.h b/thirdparty/embree/common/math/emath.h
index 22a89a7669..64e230d1db 100644
--- a/thirdparty/embree/common/math/emath.h
+++ b/thirdparty/embree/common/math/emath.h
@@ -15,6 +15,12 @@
 #if defined(__ARM_NEON)
 #include "../simd/arm/emulation.h"
 #else
+// -- GODOT start --
+#if defined(_M_ARM64)
+// Windows MSVC allows for *mmintrin.h on Arm64 but only with USE_SOFT_INTRINSICS.
+#define USE_SOFT_INTRINSICS
+#endif
+// -- GODOT end --
 #include <emmintrin.h>
 #include <xmmintrin.h>
 #include <immintrin.h>
diff --git a/thirdparty/embree/common/sys/intrinsics.h b/thirdparty/embree/common/sys/intrinsics.h
index f5074bb29d..23a99d02d3 100644
--- a/thirdparty/embree/common/sys/intrinsics.h
+++ b/thirdparty/embree/common/sys/intrinsics.h
@@ -12,6 +12,13 @@
 #if defined(__ARM_NEON)
 #include "../simd/arm/emulation.h"
 #else
+// -- GODOT start --
+// -- GODOT end --
+#if defined(_M_ARM64)
+// Windows MSVC allows for immintrin.h on Arm64 but only with USE_SOFT_INTRINSICS.
+#define USE_SOFT_INTRINSICS
+#endif
+// -- GODOT end --
 #include <immintrin.h>
 #if defined(__EMSCRIPTEN__)
 #include "../simd/wasm/emulation.h"
@@ -89,15 +96,20 @@ namespace embree
 #endif
   }
   
-#if defined(__X86_64__) || defined (__aarch64__)
-  __forceinline size_t bsf(size_t v) {
+// -- GODOT start --
+//#if defined(__X86_64__) || defined (__aarch64__)
+//  __forceinline size_t bsf(size_t v) {
+  __forceinline uint64_t bsf(uint64_t v) {
+// -- GODOT end --
 #if defined(__AVX2__) 
     return _tzcnt_u64(v);
 #else
     unsigned long r = 0; _BitScanForward64(&r,v); return r;
 #endif
   }
-#endif
+// -- GODOT start --
+//#endif
+// -- GODOT end --
   
   __forceinline int bscf(int& v) 
   {
@@ -113,14 +125,19 @@ namespace embree
     return i;
   }
   
-#if defined(__X86_64__) || defined (__aarch64__)
-  __forceinline size_t bscf(size_t& v) 
+// -- GODOT start --
+//#if defined(__X86_64__) || defined (__aarch64__)
+//  __forceinline size_t bscf(size_t& v)
+  __forceinline uint64_t bscf(uint64_t& v)
+// -- GODOT end --
   {
-    size_t i = bsf(v);
+    uint64_t i = bsf(v);
     v &= v-1;
     return i;
   }
-#endif
+// -- GODOT start --
+//#endif
+// -- GODOT end --
   
   __forceinline int bsr(int v) {
 #if defined(__AVX2__)  && !defined(__aarch64__)
@@ -138,15 +155,20 @@ namespace embree
 #endif
   }
   
-#if defined(__X86_64__) || defined (__aarch64__)
-  __forceinline size_t bsr(size_t v) {
+// -- GODOT start --
+//#if defined(__X86_64__) || defined (__aarch64__)
+//  __forceinline size_t bsr(size_t v) {
+  __forceinline uint64_t bsr(uint64_t v) {
+// -- GODOT end --
 #if defined(__AVX2__) 
     return 63 -_lzcnt_u64(v);
 #else
     unsigned long r = 0; _BitScanReverse64(&r, v); return r;
 #endif
   }
-#endif
+// -- GODOT start --
+//#endif
+// -- GODOT end --
   
   __forceinline int lzcnt(const int x)
   {
diff --git a/thirdparty/embree/common/sys/platform.h b/thirdparty/embree/common/sys/platform.h
index d4a9b9e119..a8bcc356ec 100644
--- a/thirdparty/embree/common/sys/platform.h
+++ b/thirdparty/embree/common/sys/platform.h
@@ -58,7 +58,10 @@
 #endif
 
 /* detect 64 bit platform */
-#if defined(__X86_64__) || defined(__aarch64__)
+// -- GODOT start --
+//#if defined(__X86_64__) || defined(__aarch64__)
+#if UINTPTR_MAX == UINT64_MAX
+// -- GODOT end --
 #define __64BIT__
 #endif
 
